build/common/installer/scripts/tomlparser-agent-config.rb

```diff
@@ -57,6 +57,28 @@
 @fbitTailMemBufLimitMBs = 0
 @fbitTailIgnoreOlder = ""
 
+# Checking to see if this is the daemonset or replicaset to parse config accordingly
+@controllerType = ENV["CONTROLLER_TYPE"]
+@daemonset = "daemonset"
+# Checking to see if container is not prometheus sidecar.
+# CONTAINER_TYPE is populated only for prometheus sidecar container.
+@containerType = ENV["CONTAINER_TYPE"]
+
+@promFbitChunkSize = 0
+@promFbitBufferSize = 0
+@promFbitMemBufLimit = 0
+
+# set defaults
+if !@controllerType.nil? && !@controllerType.empty? && @controllerType.strip.casecmp(@daemonset) == 0
```

Comment:
since multiple places this condition has been used so this can be wrapped into the helper function for example isDaemonSet.